modules/margin-protocol/src/lib.rs

```diff
@@ -339,10 +330,10 @@ impl<T: Trait> Module<T> {
 
 	fn _close_position(who: &T::AccountId, position_id: PositionId, price: Option<Price>) -> DispatchResult {
 		let position = Self::positions(position_id).ok_or(Error::<T>::PositionNotFound)?;
-		let index = Self::positions_by_trader(who, position.pool)
-			.iter()
-			.position(|id| *id == position_id)
-			.ok_or(Error::<T>::PositionNotOpenedByTrader)?;
+		ensure!(
+			Self::positions_by_trader(who, (position.pool, position_id)).is_some(),
```

Comment:
In general, should use `exists` to check if a value exists so it does not need to decode the value.
In this case, the value is `()` so there are no overhead.